test_suite/generic/test_generic.py

```diff
@@ -181,8 +182,12 @@ def test_sshd(self, host):
             sshd = host.service('sshd')
             assert sshd.is_running, 'ssh.service is not active'
 
-            if not host.file('/etc/ssh/sshd_config').contains('PasswordAuthentication no') and \
-               not host.file('/etc/ssh/sshd_config').contains('#PasswordAuthentication yes'):
+            sshd_config_string = host.file('/etc/ssh/sshd_config').content_string
+            pwdAuthNo = 'PasswordAuthentication no'
+            pwdAuthYes = 'PasswordAuthentication yes'
+
+            if not re.match(pwdAuthNo, sshd_config_string) and \
```

Comment:
Actually, how the code is done now, then we will have a false positive in the case the following line is present:
`PasswordAuthentication yes`

I think we can replace the `if` by an `assert` to follow what is being done in the other tests. In that case, the condition statements become "positive" and you avoid negations.

```
assert (re.match(pwdAuthNo, sshd_config_string) and 
    not re.match(pwdAuthYes, sshd_config_string)) or \
    re.match(f'#[ ]?{pwdAuthYes}|#[ ]+{pwdAuthYes}', sshd_config_string), \
    'Password authentication should be disabled'
```